include/RAJA/policy/cuda/raja_cudaerrchk.hpp

```diff
@@ -25,19 +25,94 @@
 #if defined(RAJA_ENABLE_CUDA)
 
 #include <iostream>
+#include <utility>
+#include <algorithm>
+#include <tuple>
+#include <array>
+#include <string_view>
 #include <string>
+#include <sstream>
+#include <stdexcept>
 
 #include <cuda.h>
 #include <cuda_runtime.h>
 
+#include "camp/defines.hpp"
+#include "camp/helpers.hpp"
+
 #include "RAJA/util/macros.hpp"
 
+#include "cub/util_type.cuh"
+
+namespace camp
+{
+
+namespace experimental
+{
+
+template < >
+struct StreamInsertHelper<CUDA_DIM_T&>
+{
+  CUDA_DIM_T& m_val;
+
+  std::ostream& operator()(std::ostream& str) const
+  {
+    return str << "{" << m_val.x
+               << "," << m_val.y
+               << "," << m_val.z
+               << "}";
+  }
+};
+///
+template < >
+struct StreamInsertHelper<CUDA_DIM_T const&>
+{
+  CUDA_DIM_T const& m_val;
+
+  std::ostream& operator()(std::ostream& str) const
+  {
+    return str << "{" << m_val.x
+               << "," << m_val.y
+               << "," << m_val.z
+               << "}";
+  }
+};
+
+template < typename R >
+struct StreamInsertHelper<::cub::DoubleBuffer<R>&>
+{
+  ::cub::DoubleBuffer<R>& m_val;
+
+  std::ostream& operator()(std::ostream& str) const
+  {
+    return str << "{" << m_val.Current() << "," << m_val.Alternate() << "}";
+  }
+};
+///
+template < typename R >
+struct StreamInsertHelper<::cub::DoubleBuffer<R> const&>
+{
+  ::cub::DoubleBuffer<R> const& m_val;
+
+  std::ostream& operator()(std::ostream& str) const
+  {
+    // Can't get current and alternate as they are non-const functions
+    return str << "{?,?}";
+  }
+};
+
+}  // namespace experimental
+
+}  // namespace camp
+
+
 namespace RAJA
 {
 
 ///
 ///////////////////////////////////////////////////////////////////////
 ///
+/// DEPRECATED
```

Comment:
Are we using this within RAJA or is this deprecation for users who may be using the RAJA cudaAssert() method?

Reply:
We are no longer using this, I deprecated it in case any users of RAJA were using this.